src/login/logind-button.c

```diff
@@ -214,18 +279,19 @@ static int button_dispatch(sd_event_source *s, int fd, uint32_t revents, void *u
 
         if (ev.type == EV_KEY && ev.value > 0) {
 
+                keycode_modifier_mask = button_get_keycode_modifier_mask(ev.code);
```

Comment:
why call this here btw? you only need it in the `default:` branch of the switch statement below, hence move it there?